plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/plan/CubridPlanNode.java

```diff
@@ -149,111 +181,183 @@ private String getMethodTitle(@NotNull String method) {
             }
         }
     }
 
-    private void parseObject(@NotNull List<String> segments) {
-        if (!segments.isEmpty()) {
-            String[] removes = segments.remove(0).split(OPTIONS_SEPARATOR);
-            nodeProps.put(removes[0], removes[1].trim());
-            if (removes[0].equals(COST) || segments.isEmpty()) {
-                return;
+    private boolean subNode(CubridPlanNode node, String key, String value) {
+        if (value.contains(" AND ")) {
+
+            Matcher m = subNodePattern.matcher(value);
+            int count = 1;
+            while (m.find()) {
+                node.addNested(false, String.format("%s %s:%s", key, count, m.group()));
+                count++;
             }
-            String key = segments.get(0).split(OPTIONS_SEPARATOR)[0];
-            if (nodeProps.containsKey(key) || removes[0].equals("subplan")) {
-                addNested(removes[1].trim(), segments);
-                parseObject(segments);
-            } else if (key.equals(CLASS)) {
-                if (!removes[0].equals("Query plan")) {
-                    addNested(removes[1].trim(), segments);
+            return true;
+        }
+        return false;
+    }
+
+
+    @Nullable
+    private String getTermValue(String value) {
+        if (CommonUtils.isNotEmpty(value)) {
+            if (value.contains("node[")) {
+                Matcher m = termPattern.matcher(value);
+                if (m.find()) {
+                    return value.replace(m.group(), classNode.get(m.group()));
                 }
-                parseObject(segments);
             } else {
-                parseObject(segments);
+                String termValue = terms.get(value);
+                if (CommonUtils.isNotEmpty(termValue)) {
+                    return termValue.split(" \\(sel ")[0];
+                }
             }
         }
+        return null;
+
     }
 
     @Nullable
-    private String getTermExtra(boolean isTerm) {
-        
-        String termValue = terms.get(term);
-        if(CommonUtils.isNotEmpty(termValue)) {
-            String[] values = termValue.split(" \\(sel");
-            if (isTerm)
-                return values[0].trim();
-            else
-                return "(sel" + values[1].trim();
+    private String getExtraValue(String value) {
+        String extraValue = terms.get(value);
+        if (CommonUtils.isNotEmpty(extraValue)) {
+            return "(sel " + extraValue.split(" \\(sel ")[1];
         }
         return null;
     }
 
-    @Nullable
-    private String getNameOrTotal(boolean isName) {
-        String classNodeValue = classNode.get(nodeName);
-        if(CommonUtils.isNotEmpty(classNodeValue)){
-            Pattern p;
-            if (isName) {
-                p = Pattern.compile("\\w+ \\w+");
-            } else {
-                p = Pattern.compile("\\w+\\/\\w+");
-            }
-            Matcher m = p.matcher(classNodeValue);
+    private void getNameValue(String value) {
```

Comment:
These names are misleading. The `get` prefix indicates that it's a getter method, but in reality, it doesn't return anything and just extracts string segments. Please call it differently.